src/ObjectUtils/CMakeLists.txt

```diff
@@ -31,7 +33,9 @@ target_sources(
         WindowsBuildIdUtils.cpp)
 
 if (NOT WIN32)
-target_sources(ObjectUtils PRIVATE LinuxMap.cpp)
+target_sources(ObjectUtils PRIVATE
+        ReadMaps.cpp
+        ReadModules.cpp)
 else()
```

Comment:
Reading these `if()` made me think this component might not be a good location for this maps related code. How about a new component for the files ReadMaps and ReadModules? Then the new component could just be added for Linux in our global CMakeLists.txt. I think all components who actually use this maps functionality are also Linux only. (Which makes a lot of sense)

I know this is a biggish ask, but I think it would be worth it. You can then also get rid of the `#ifdef __linux` in ReadModules.h

Reply:
I had considered this, but I can't think of a single solution that I really like. Let's discuss offline so that I can explain what I mean.